crates/fluvio-spu/src/services/public/mod.rs

```diff
@@ -1,30 +1,143 @@
 mod api_versions;
-mod service_impl;
 mod produce_handler;
 mod fetch_handler;
 mod offset_request;
+mod offset_update;
 mod stream_fetch;
 
-use tracing::info;
+use std::sync::Arc;
+use async_trait::async_trait;
+use tracing::{info, debug, trace, instrument};
+use futures_util::StreamExt;
 
-use fluvio_service::FluvioApiServer;
-use service_impl::PublicService;
+use fluvio_socket::{FluvioSocket, SocketError};
+use fluvio_service::{FluvioApiServer, FluvioService, ConnectInfo, call_service};
 use fluvio_spu_schema::server::SpuServerRequest;
 use fluvio_spu_schema::server::SpuServerApiKey;
+use fluvio_types::event::StickyEvent;
 
 use crate::core::DefaultSharedGlobalContext;
+use self::api_versions::handle_api_version_request;
+use self::produce_handler::handle_produce_request;
+use self::fetch_handler::handle_fetch_request;
+use self::offset_request::handle_offset_request;
+use self::offset_update::handle_offset_update;
+use self::stream_fetch::StreamFetchHandler;
 pub use stream_fetch::publishers::StreamPublishers;
 
-pub(crate) type PublicApiServer =
-    FluvioApiServer<SpuServerRequest, SpuServerApiKey, DefaultSharedGlobalContext, PublicService>;
+pub(crate) type SpuPublicServer = FluvioApiServer<
+    SpuServerRequest,
+    SpuServerApiKey,
+    DefaultSharedGlobalContext,
+    SpuPublicService,
+>;
 
-// start server
-pub fn create_public_server(addr: String, ctx: DefaultSharedGlobalContext) -> PublicApiServer {
-    info!(
-        "starting SPU: {} at public service at: {}",
-        ctx.local_spu_id(),
-        addr
-    );
+#[derive(Debug)]
+pub struct SpuPublicService {
+    _0: (), // Prevent construction
+}
+
+impl SpuPublicService {
+    pub fn new() -> Self {
+        SpuPublicService { _0: () }
+    }
+
+    pub fn server(addr: String, ctx: DefaultSharedGlobalContext) -> SpuPublicServer {
+        info!(
+            spu_id = ctx.local_spu_id(),
+            %addr,
+            "Starting SPU public service:",
+        );
+
+        FluvioApiServer::new(addr, ctx, SpuPublicService::new())
```

Comment:
This seems like pointless indirection.  Shouldn't better to wrap `FluvioApiServer` ?

Reply:
This is exactly the same code as what used to be called `create_public_server`, I just put it as an associated function on `SpuPublicService` as a convenience

Reply:
`_0` is not standard Rust idiom.  Still don't see why need to be wrapped.

Reply:
I have seen it used before, different folks use different names but the pattern is the same:
https://rust-unofficial.github.io/patterns/idioms/priv-extend.html#alternative-private-fields-for-structs